pkg/raft/term_cache_test.go

```diff
@@ -0,0 +1,396 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package raft
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	pb "github.com/cockroachdb/cockroach/pkg/raft/raftpb"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTermCacheInitialization(t *testing.T) {
+	tc := NewTermCache(8, entryID{term: 5, index: 10})
+
+	// Verify basic properties after initialization
+	require.Len(t, tc.cache, 1)
+	require.Equal(t, entryID{term: 5, index: 10}, tc.firstEntry())
+	require.Equal(t, uint64(10), tc.cache[0].index)
+	require.Equal(t, 8, cap(tc.cache))
+}
+
+func TestTermCacheTerm(t *testing.T) {
+	// Create a term cache with initial entry at term=5, index=10
+	initialID := entryID{term: 5, index: 10}
+	tc := NewTermCache(4, initialID)
+
+	// Create a log slice with entries that have different term patterns
+	// This creates entries at indices 11-14 with terms [5,6,6,7]
+	leadSlice := initialID.append(5, 6, 6, 7)
+	tc.truncateAndAppend(leadSlice.LogSlice)
+
+	// Test term lookup for various indices
+	for _, tt := range []struct {
+		name     string
+		index    uint64
+		wantTerm uint64
+		wantOK   bool
+	}{
+		{name: "below cache", index: 9, wantTerm: 0, wantOK: false},
+		{name: "initial entry", index: 10, wantTerm: 5, wantOK: true},
+		{name: "same term continuation", index: 11, wantTerm: 5, wantOK: true},
+		{name: "term flip", index: 12, wantTerm: 6, wantOK: true},
+		{name: "same term", index: 13, wantTerm: 6, wantOK: true},
+		{name: "another term flip", index: 14, wantTerm: 7, wantOK: true},
+		// NB: Although index 15 is not inserted to the cache,
+		// we still return term = 7. Callers should make sure
+		// that index <= lastIndex when calling term(index).
+		{name: "beyond cache", index: 15, wantTerm: 7, wantOK: true},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			term, ok := tc.term(tt.index)
+			require.Equal(t, tt.wantTerm, term, "index %d", tt.index)
+			require.Equal(t, tt.wantOK, ok, "index %d", tt.index)
+		})
+	}
+}
+
+func TestTermCachePopulatedTruncateAndAppend(t *testing.T) {
+	testCases := []struct {
```

Comment:
Use the same `for _, tt := range []struct { } ...` pattern as in `TestTermCacheTerm`? Here and in other places if any.